src/Split.tsx

```diff
@@ -115,19 +115,46 @@ export class Split extends React.Component<SplitProps, ISplitState> {
   }
   getMainSizeStyle = () => {
     switch (this.state.mode) {
-      case 'minimize':
-        return '0px';
-      case 'maximize':
-        return '100%';
+      case "minimize":
+        if (this.state.minSize > 0) {
+          return `${this.state.minSize}px`
+        }
+        return "0px"
+      case "maximize":
+        if (this.state.maxSize > 0) {
+          return `${this.state.maxSize}px`
+        }
+
+        if (this.state.maxSize < 0) {
+          return `calc(100% - ${Math.abs(this.state.maxSize)}px)`
+        }
+
+        return "100%"
       default:
+        if (this.state.size <= this.state.minSize) {
+          return `${this.state.minSize}px`
+        }
+
+        if (this.state.size >= this.state.maxSize) {
+          return `${this.state.maxSize}px`
+        }
+
         if (this.state.size === -1) {
-          return 'auto';
+          if (this.state.minSize > 0) {
+            return `${this.state.minSize}px`
+          }
+
+          return "auto"
```

Comment:
must be `auto` because we want panel to take auto size (depends on other styles)